CMakeLists.txt

```diff
@@ -102,21 +103,32 @@ endif()
 # Find OGRE2: first try to find OGRE2 built with PlanarReflections support and
 # fallback to look for OGRE2 without it. Both seems to works for gz-rendering.
 # See https://github.com/gazebosim/gz-rendering/issues/597
+
+# Build the list of OGRE2 components based on the options
+set(OGRE2_COMPONENTS HlmsPbs HlmsUnlit PlanarReflections)
+if(USE_OGRE2_OVERLAY)
+  list(APPEND OGRE2_COMPONENTS Overlay)
```

Comment:
The ABI checker picked up some compatibility issues. I think we need to set `GZ_RENDERING_HAVE_OGRE2_OVERLAY` to `TRUE` here as well.